lib/execution_engine2/db/MongoUtil.py

```diff
@@ -155,41 +156,28 @@ def _get_collection(
 
         return pymongo_client, mongoengine_client
 
-    @contextmanager
-    def pymongo_client(self, mongo_collection):
-        """
-        Instantiates a mongo client to be used as a context manager
-        Closes the connection at the end
-        :return:
-        """
-        self.mongo_collection = mongo_collection
-        yield self.pymongoc
```

Comment:
I've always been pretty confused about this point of this method, since it doesn't really do anything other than return the client. It doesn't actually close the client connection (which it shouldn't, so that's good) and doesn't do any context managing AFAICT. Setting the mongo collection instance variable here is what caused the problem so I decided to just delete the whole thing and simplify.